docs/scos/dev/tutorials-and-howtos/howtos/howto-process-incoming-HTTP-requests-for-order-updates.md

```diff
@@ -0,0 +1,131 @@
+
+{% info_block infoBox "Info" %}
+
+**Important point:** An incoming request handler must store not only the event context but also trigger an OMS event. It is necessary to pass control to oms:check-condition worker. If the event must affect several orders then the handler must trigger OMS event for each order.
+
+{% endinfo_block %}
+
+**Pros**
+
+* The worker is available OOTB
+* More transparency with the logic that is represented in OMS
+* Easy understand, maintain, support
+* OOTB fault-tolerance and supports retry logic
+
+**Cons**
+
+* Can run a logic in OMS Plugins only
+* Cannot trigger OMS events for the same Order
+* Extra OMS elements may make the graph more complex to understand and maintain
+
+##### Incoming request handling and passing control to the worker
+
+The incoming request handler should validate a Request structure, save it in some Storage and then trigger an OMS event. E.g. “start processing”. OMS will stop order items in the next state. On the next tick, “oms:check-condition” worker will move them forward and run subsequent commands and logic.
+
+![Incoming request handling and passing control to the worker](docs/scos/dev/drafts-dev/Incoming request handling and passing control to the worker.png)
+
+{% info_block warningBox "Warning" %}
+
+**Potential pitfall:** The worker can process the requests only when an order is in a specific OMS state. If OMS cannot apply “start processing” event to any item then it returns an error. It means that the event can’t be processed right now. The request handler should decide what to do: store it for further processing, ignore it, or return an error response to the caller.
+
+{% endinfo_block %}
+
+Another way to check the order items' status is to check the current state of all items in the Database (spy_sales_order_item_state). It can be useful in some cases. This way can be a bit faster but prone to the issue of a concurrent request and not recommended.
+
+#### Processing with a dedicated Jenkins worker
+
+The Jenkins worker listens to the Storage. When an event appears, the worker starts to process it. It is a high-level logic. Depending on requirements, the worker logic can be very different and cover quality attributes differently.
+
+| Quality attribute                                     | Suggestions                                                                                                                                                                                                     |
```

Comment:
"Suggestions" - is it Jenkins configuration suggestions (recommendations"

Reply:
it is for "Processing with a dedicated Jenkins worker" solution in general. So, if a developer chooses this solution (dedicated worker) then these suggestions may be useful.